src/views/map-view.ts

```diff
@@ -321,6 +321,42 @@ export class MapView<N extends INodeBase, E extends IEdgeBase> implements IOrbVi
           globalPoint: containerPoint,
         });
 
+        if (response.isStateChanged || response.changedSubject) {
+          this._renderer.render(this._graph);
+        }
+      } else if (event.type === 'dblclick' && this._strategy.onMouseDoubleClick) {
+        const response = this._strategy.onMouseDoubleClick(this._graph, point);
+        const subject = response.changedSubject;
+
+        if (subject) {
+          if (isNode(subject)) {
+            this._events.emit(OrbEventType.NODE_DOUBLE_CLICK, {
+              node: subject,
+              event: event.originalEvent,
+              localPoint: point,
+              globalPoint: containerPoint,
+            });
+          }
+          if (isEdge(subject)) {
+            this._events.emit(OrbEventType.EDGE_DOUBLE_CLICK, {
+              edge: subject,
+              event: event.originalEvent,
+              localPoint: point,
+              globalPoint: containerPoint,
+            });
+          }
+        } else {
+          this._events.emit(OrbEventType.MOUSE_DOUBLE_CLICK, {
+            subject,
+            event: event.originalEvent,
+            localPoint: point,
+            globalPoint: containerPoint,
+          });
+          // zoom in on double click
+          const zoom = event.target._zoom + 1;
+          event.target.setZoomAround(event.layerPoint, zoom);
+        }
```

Comment:
```suggestion
        }
        
        this._events.emit(OrbEventType.MOUSE_DOUBLE_CLICK, {
          subject,
          event: event.originalEvent,
          localPoint: point,
          globalPoint: containerPoint,
        });
        
        // zoom in on double click if no subject underneath
        if (!subject) {
          const zoom = event.target._zoom + 1;
          event.target.setZoomAround(event.layerPoint, zoom);
        }
```

The current code won't send MOUSE_DOUBLE_CLICK events (even if the subject exists) which the user might listen for. All other general mouse events send MOUSE_CLICK events for both cases if the subject is defined or not defined, so we should keep the same behavior here too.

Reply:
yes got it. Thanks !